sample_project/Assets/SampleViewer/Samples/Geometry/Geometries.cs

```diff
@@ -0,0 +1,503 @@
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+
+using Esri.ArcGISMapsSDK.Components;
+using Esri.ArcGISMapsSDK.Samples.Components;
+using Esri.ArcGISMapsSDK.Utils.GeoCoord;
+using Esri.GameEngine.Geometry;
+using Esri.HPFramework;
+using System;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+using Unity.Mathematics;
+
+public class Geometries : MonoBehaviour
+{
+	[Header("-----------ArcGIS Components-----------")]
+	private ArcGISCameraControllerComponent arcGISCameraControllerComponent;
+	private ArcGISMapComponent arcGISMapComponent;
+	private double3 lastRootPosition;
+
+	[Header("-----------Line Objects-----------")]
+	[SerializeField] public float InterpolationInterval = 100;
+	[SerializeField] public GameObject InterpolationMarker;
+	[SerializeField] public GameObject Line;
+	[SerializeField] public GameObject LineMarker;
+	[SerializeField] float MarkerHeight = 200f;
+	private List<GameObject> featurePoints = new List<GameObject>();
+	private double geodeticAreaEnvelope = 0;
+	private double geodeticAreaPolygon = 0;
+	private double geodeticDistance = 0;
+	private List<GameObject> lastToStartInterpolationPoints = new List<GameObject>();
+	private const float LineWidth = 5f;
+	private LineRenderer lineRenderer;
+	private const float RaycastHeight = 5000f;
+	private ArcGISPoint startPoint;
+	private Stack<GameObject> stops = new Stack<GameObject>();
+
+	[Header("-----------Units-----------")]
+	private ArcGISAreaUnit currentAreaUnit = new ArcGISAreaUnit(ArcGISAreaUnitId.SquareMiles);
+	private ArcGISLinearUnit currentLinearUnit = new ArcGISLinearUnit(ArcGISLinearUnitId.Miles);
+	private string unitText;
+
+	[Header("-----------State Management-----------")]
+	private bool isDragging = false;
+	private bool isEnvelopeMode = false;
+	private bool isPolygonMode = false;
+	private bool isPolylineMode = true;
+
+	[Header("-----------UI-----------")]
+	[SerializeField] public Button ClearButton;
+	[SerializeField] public Button[] ModeButtons;
+	[SerializeField] public TMP_Text result;
+	[SerializeField] public Button[] UnitButtons;
+
+	private void Start()
+	{
+		arcGISMapComponent = FindObjectOfType<ArcGISMapComponent>();
+		arcGISCameraControllerComponent = FindObjectOfType<ArcGISCameraControllerComponent>();
+
+		lineRenderer = Line.GetComponent<LineRenderer>();
+		lineRenderer.widthMultiplier = LineWidth;
+		lastRootPosition = arcGISMapComponent.GetComponent<HPRoot>().RootUniversePosition;
+
+		ClearButton.onClick.AddListener(delegate
+		{
+			ClearLine();
+		});
+
+		ModeButtons[0].interactable = false;
+		UnitButtons[0].interactable = false;
+	}
+
+	private void Update()
+	{
+		if (EventSystem.current.IsPointerOverGameObject())
+		{
+			// Block 3D raycasts when mouse is over UI
+			return;
+		}
+
+		if (isEnvelopeMode)
+		{
+			if (Input.GetKey(KeyCode.LeftShift))
```

Comment:
Can we switch the input over to the new input system to keep it consistent with all the other unity samples?

Reply:
We could. Although geocoding and routing use the old input system too (see images below). My thinking is that if something enhances UX we should strive to do it, if users don't get impacted, we don't need to transfer all these samples to the new input system. 

![image](https://github.com/user-attachments/assets/82d9039c-53b5-40cf-b4ec-daabac127bbe)
![image](https://github.com/user-attachments/assets/2bf08de9-afc9-4aa4-9b21-d5f21509c603)

Reply:
I think new samples should use the new input system since it is what I’d recommended by unity. 

Additionally porting over the geocoding and routing to the new input system would improve quality of life for all samples since it will entirely remove the old input and we would be reusing input actions already created (created for hit test). Unity highly recommends using the new input system despite the old one still existing.